README.md

```diff
@@ -1270,19 +1271,24 @@ this workflow.
 For a simple example of using PyTorch in a Python Backend model, see the
 [AddSubNet PyTorch example](#addsubnet-in-pytorch).
 
-### Determinism and Reproducibility
+### PyTorch Determinism 
 
 When running PyTorch code, you may notice slight differences in output values
-across runs or across servers depending on hardware, system load, driver, etc. 
+across runs or across servers depending on hardware, system load, driver, or even
+batch size. These differences are generally related to the selection of CUDA
+kernels used to execute the operations, based on the factors mentioned.
+
 For most intents and purposes, these differences aren't large enough to affect
 a model's final prediction. However, to understand where these differences come 
 from, see this [doc](https://pytorch.org/docs/stable/notes/randomness.html).
 
 On Ampere devices and later, there is an optimization related to
-FP32 operations called TensorFlow32 (TF32). Typically this optimization will
-improve overall performance at the cost of minor precision loss, but similarly
-this precision loss is acceptable for most model predictions. For more info on
-TF32 in PyTorch and how to enable/disable it as needed, see 
+FP32 operations called 
+[TensorFloat32 (TF32)](https://blogs.nvidia.com/blog/2020/05/14/tensorfloat-32-precision-format/).
+Typically this optimization will improve overall performance at the cost of
+minor precision loss, but similarly this precision loss is acceptable for most
+model predictions. For more info on TF32 in PyTorch and how to enable/disable
+it as needed, see 
```

Comment:
I think what we want to highlight here is that TF32 is enabled by default (need to confirm) in PyTorch when using PyTorch backend but it is disabled in the upstream PyTorch and that's why the results could be different. https://pytorch.org/docs/stable/notes/cuda.html#tensorfloat-32-tf32-on-ampere-devices

Reply:
Makes sense. Resolving.